Formula/i/icann-rdap.rb

```diff
@@ -0,0 +1,32 @@
+class IcannRdap < Formula
+  desc "Full-rich client for the Registry Data Access Protocol (RDAP) sponsored by ICANN"
+  homepage "https://github.com/icann/icann-rdap/wiki"
+  url "https://github.com/icann/icann-rdap/archive/refs/tags/v0.0.21.tar.gz"
+  sha256 "252b112776fae0160f539e20b70ff24b6f2bea7551c9476ccd6f7651c7b861d0"
+  license any_of: ["Apache-2.0", "MIT"]
+
+  depends_on "pkgconf" => :build
+  depends_on "rust" => :build
+  depends_on "openssl@3"
+
+  conflicts_with "rdap", because: "openrdap also ships a rdap binary"
+
+  def install
+    # the std_cargo_args has a path argument which cannot be repeated.
+    # therefore, we remove that path argument and put one pointing to the
+    # icann-rdap-cli sub-package.
+    @cargo_args = std_cargo_args.reject { |a| a.include?("--path") }
+    @cargo_args << "--path=icann-rdap-cli"
+    # install only rdap and rdap-test and not rdap-srv*
+    system "cargo", "install", "--bin=rdap", *@cargo_args
+    system "cargo", "install", "--bin=rdap-test", *@cargo_args
+  end
+
+  test do
+    # Test rdap binary executes
+    system "#{bin}/rdap", "-h"
+
+    # Test rdap-test binary executes
+    system "#{bin}/rdap-test", "-h"
```

Comment:
The test should include version check and real use case, for example
```suggestion
    assert_match version.to_s, shell_output("rdap -V")
    assert_match version.to_s, shell_output("rdap-test -V")

    system "#{bin}/rdap", "icann.org"
    system "#{bin}/rdap-test", "--skip-v6", "icann.org"
```

Reply:
Unfortunately, real use cases require network access and rely on third-party infrastructure to be on-line. Network-access is probably not that controversial as homebrew probably requires that anyway. But in the test cases above, both of those are hitting IANA and PIR (and Google's DNS for rdap-test). While all of those are very resilient and robust services, is it acceptable for the tests to rely on them?

Reply:
I think that is ok, we have several formulae to test their functions through the network, and they use the famous domains such as github.com, example.com, google.com and so on.